app/src/main/java/com/yml/womensafety/navigationdrawer/home/HomePageActivity.kt

```diff
@@ -0,0 +1,109 @@
+        databaseReference = firebaseDatabase!!.reference.child("name")
+        val user = auth.currentUser
+        val userEmail = user?.email.toString()
+        val userReference = databaseReference?.child(user?.uid!!)
+        userReference?.addValueEventListener(object : ValueEventListener {
+            @SuppressLint("SetTextI18n")
+            override fun onDataChange(snapshot: DataSnapshot) {
+                val userFullName = snapshot.child("fullName").value.toString()
+                //Toast.makeText(view.context,"Welcome $userFullName",Toast.LENGTH_LONG).show()
+                Snackbar.make(findViewById(android.R.id.content),"Welcome $userFullName", Snackbar.LENGTH_LONG).show()
+            }
+
+            override fun onCancelled(error: DatabaseError) {
+
+            }
+
+        })
+        cvEscapeThreat.setOnClickListener {
+            tipsToEscapeFragment = TipsToEscapeFragment()
+            supportFragmentManager
+                .beginTransaction()
+                .replace(R.id.flHomePage,tipsToEscapeFragment)
+                .setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN)
+                .addToBackStack(null)
+                .commit()
+
+        }
+        cvVideosForSelfDefense.setOnClickListener {
+            selfDefenseVideoFragment = SelfDefenseVideoFragment()
+            supportFragmentManager
+                .beginTransaction()
+                .replace(R.id.flHomePage,selfDefenseVideoFragment)
+                .setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN)
+                .addToBackStack(null)
+                .commit()
+        }
+        bottomNavigationView.setOnNavigationItemReselectedListener {
+            when(it.itemId){
+                R.id.navHome -> {
+                    startActivity(Intent(this, HomePageActivity::class.java))
+//                    homePageFragment = HomePageFragment()
+//                    supportFragmentManager
+//                        .beginTransaction()
+//                        .replace(R.id.flHomePage, homePageFragment)
+//                        .setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN)
+//                        .addToBackStack(null)
+//                        .commit()
+                }
+                R.id.navLogout -> {
+                    auth.signOut()
+                    startActivity(Intent(this, LoginActivity::class.java))
+                    finish()
+                }
+                R.id.navContacts -> {
+                    contactsFragment = ContactsFragment()
+                    supportFragmentManager
+                        .beginTransaction()
+                        .replace(R.id.flHomePage,contactsFragment)
+                        .setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN)
+                        .commit()
+
+                }
+            }
+            true
+        }
+
+    }
+
+    override fun onBackPressed() {
+        super.onBackPressed()
```

Comment:
OnBackPress will handle finishing activity automatically. No need to override it.

Reply:
Done